osp/views/github_stat.py

```diff
@@ -0,0 +1,56 @@
+from rest_framework import status, viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from osp.models import UserInformation
+from osp.models.github_stat import GithubStat
+from osp.serializers.github_stat import GithubStatSerializer
+from osp.utils.github_api_client import GithubClient
+
+
+class GithubStatView(viewsets.ModelViewSet):
+    permission_classes = [
+        IsAuthenticated,
+    ]
+    serializer_class = GithubStatSerializer
+    github_client = GithubClient()
+    queryset = GithubStat.objects.all()
+
+    def get_queryset(self):
+        user_id = self.request.query_params.get("user_id", None)
+        if user_id is None:
+            user = self.request.user
+            user_id = user.id
+        return self.queryset.filter(user_information__user_id=user_id)
+
+    def create(self, request):
+        user = self.request.user
+        try:
+            user_information = UserInformation.objects.get(user__id=user.id)
+        except UserInformation.DoesNotExist:
+            return Response(
+                {
+                    "message": "User information does not exist. You must create it in order to fetch the GitHub "
+                    "details!"
+                },
+                status=status.HTTP_404_NOT_FOUND,
+            )
+        github_stat = GithubStat.objects.get(user_information__user_id=user.id)
```

Comment:
What if there is no object for that user in GithubStat?

Reply:
I am getting errors due to this, please handle errors also.

Reply:
Sorry @codesankalp for the late response. I got stuck with some things, so I was less active. I'll take a look.